lib/_tls_common.js

```diff
@@ -196,6 +196,18 @@ exports.createSecureContext = function createSecureContext(options, context) {
     c.context.setFreeListLength(0);
   }
 
+  if (typeof options.clientCertEngine === 'string') {
+    if (c.context.setClientCertEngine)
+      c.context.setClientCertEngine(options.clientCertEngine);
+    else
+      throw new Error('Custom engines not supported by this OpenSSL');
```

Comment:
Would prefer that new errors use internal/errors

Reply:
Agreed, but doing so effectively will require scope-creep-like cleanup in `internal/errors.js` that should be a separate PR. I'll open that PR and probably label this one as blocked until that one lands.

Reply:
Meh, we don't really need any changes to `internal/errors.js` for this. I'll implement this nit now. Nothing's blocking it, I don't think.

Reply:
OK, done. Still need to add tests for the new errors. Coming shortly....